### DISCUSSION TITLE
Support for the Serbian language

### DISCUSSION THREAD

**Discussion by ZeljkoBG** [organicmaps/organicmaps#9858] [Translations]
Please add my Cyrillic translation to the application - https://github.com/user-attachments/files/17948072/strings.txt

After checking in the application, the translation will be corrected, if necessary.
Then I will continue with the translation of the other lang files.

**Comment by map-per** [comment]
@dobridabar opened #9857 for Serbian translations at the same time you opened this discussion.

**Reply by ZeljkoBG** [reply]
I don't know. I didn't see it. I was told to open a topic so I did. I don't have time for other posts. I spent a lot of time trying to get an explanation of what to translate...

And I don't know what I should do now. Either leave me alone or merge the topic.

**Comment by ZeljkoBG** [comment]
@biodranik , @zyphlar 
I translated [types_strings.txt](https://github.com/user-attachments/files/18141534/types_strings.txt) 
Is it possible to add a Serbian translation in the next version?

**Reply by zyphlar** [reply]
@ZeljkoBG thank you for this work! Are you able to open a pull request or do you need someone to do it for you? I'm surprised the other PR was merged so quickly despite being unfinished. Maybe we can get your file merged with a PR.

**Reply by ZeljkoBG** [reply]
Please, I don't know how to open that "pull request". I don't even know what 'PR' is.

**Reply by zyphlar** [reply]
Ok, I'll open one for you

**Reply by ZeljkoBG** [reply]
Please open that "pull request"  for **types_strings.txt** and **strings.txt**

**Reply by zyphlar** [reply]
Did you intend to make these changes @ZeljkoBG ?

```
-    sk = Vysoká škola
+    sk = Факултет

-    sv = Lokalitet
+    sv = Локација

```

**Reply by biodranik** [reply]
@zyphlar reviewing it in the PR may be more convenient )

**Reply by zyphlar** [reply]
@biodranik preparing the PR, but attempting to not have to force-push and regenerate ten times and waste your time

**Reply by biodranik** [reply]
An easier approach is to avoid regen at all, it complicates a bit the review and modifications. Regen can either be added later or after merging the change, that's not a big deal.

**Reply by ZeljkoBG** [reply]
If I understand you, I made a mistake and changed the translation in another language. I apologize if I did that. Correct the error.

Correct mistakes in Serbian. It is correct
  [type.place.locality]
    sr = Локалитет

  [type.amenity.college]
    sr = Факултет

**Comment by zyphlar** [comment]
Opened #9869 for strings.txt (conflict with existing) and #9871 for types_strings.txt

@ZeljkoBG one easy way to make a pull request and have less delays is to simply click the Edit button on the file you want to edit in GitHub. It will handle the pull request creation process for you easily.

**Reply by zyphlar** [reply]
@dobridabar thank you for your contributions with Serbian! Maybe future work can be divided with Zeljko here so it's done faster with less overlap?

**Reply by dobridabar** [reply]
@zyphlar aboslutely, I am glad that there are more people interested to do the translation to Serbian.

I was very surprised to find out that both @ZeljkoBG and I created PRs for translating same file at the same day, just couple of hours apart.

Following the instructions in TRANSLATIONS.md, I anounced my interest to help with Serbian translations at 9th November 2024, here:
https://github.com/orgs/organicmaps/discussions/8538#discussioncomment-11193864

I didn't see that anyone else (ever) anounced before, so my guess was the only one interested in doing the translations, so I started working alone.

Did @ZeljkoBG already expressed his intention to translate app to Serbian? If yes, when was that? @ZeljkoBG, @biodranik, @zyphlar

**Reply by zyphlar** [reply]
Nov 20. Sorry for the confusion! 
https://github.com/orgs/organicmaps/discussions/9702

**Comment by ZeljkoBG** [comment]
@zyphlar 
I translated https://github.com/organicmaps/organicmaps/blob/master/android/app/src/fdroid/play/listings/sr/full-description.txt

[full-description.txt](https://github.com/user-attachments/files/18168091/full-description.txt)

**Comment by ZeljkoBG** [comment]
@zyphlar if I understood/did it right, then here is this translation. And if it's not good, then nothing. Let's forget about this file.
[categories_cuisines.txt](https://github.com/user-attachments/files/18168994/categories_cuisines.txt)

**Comment by ZeljkoBG** [comment]
To www.openstreetmap.org contacted me @dobridabar . 
I think we've figured out what and how to fix it in SRB translation. He will continue to do as we agreed.
**He has full consent from me.**

And I, myself, will do the new TTS text he announced @zyphlar

**Comment by zyphlar** [comment]
Thank you @ZeljkoBG ! Are you able to click the Edit button on those files and open your own pull request? That way you can manage any changes and approvals without waiting for me.

https://docs.github.com/en/repositories/working-with-files/managing-files/editing-files#editing-files-in-another-users-repository

**Comment by ZeljkoBG** [comment]
I have been on the computer since 1997. For 7 years he received a salary from working on a computer. Translated everything that can be translated from GPS navigation.
I can't manage, nor do I understand a million options to set up one TXT file. I'm sick of github.

@zyphlar  I tried to add  "full-description.txt". Please see if I did it right.
I don't know/don't understand how to add 'categories_cuisines.txt'
Please do it because I don't know.

**Reply by zyphlar** [reply]
@ZeljkoBG it's always frustrating to learn new things but you're halfway there! Remember it's complex because it's not just a text file, it's a process for approving precise changes in a bunch of text that's always changing ;)

I think if you go here there may be a green Create Pull Request button you can click.

https://github.com/ZeljkoBG/organicmaps/commit/f0018286fe33e50e707dec5db3e658652d0ccfc2

If not, go here and click the Create Pull Request button and choose ZeljkoBG/organicmaps patch-13

Https://github.com/organicmaps/organicmaps/

**Comment by ZeljkoBG** [comment]
@pastk, @zyphlar @dobridabar 

https://github.com/organicmaps/organicmaps/pull/9925#pullrequestreview-2525445223

Add translation 
[release-notes.txt](https://github.com/user-attachments/files/18275328/release-notes.txt)
[short-description.txt](https://github.com/user-attachments/files/18275355/short-description.txt)
[title-google.txt](https://github.com/user-attachments/files/18275374/title-google.txt)
[title.txt](https://github.com/user-attachments/files/18275386/title.txt)

**Reply by dobridabar** [reply]
Added to PR #10014

**Comment by ZeljkoBG** [comment]
@zyphlar I downloaded the BETA version. There are a few corrections for now. Please report/fix bugs.
string
My places - is in English. The translation is: 
Моја места

Not added  "sr" in [pref_display_kayak_title] and  [pref_display_kayak_text]

trans.

[pref_display_kayak_title]
sr = Претрага хотела на Kayak.com

[pref_display_kayak_title] 
sr = Линкови "Фотографије, рецензије, резервације" приказују се за хотеле повезане са Kayak.com. Бонус добијен, за сваку резервацију, преко ових линкова иде за развој Organic Maps.



+++++++++++++++++++++++++++
Incorrect translation
Correct translation
**********************************************
Саобраћајне камере
Прекршајне камере

Упозорење на прекорачење брзине
Упозорити код се прекорачи брзина

Омогући снимање лог фајла
Омогућите снимање лог фајла

Повећај слова на мапи
Повећајте слова на мапи

Сачувај мапе у
Место за чување мапа

Држи укључен екран
Екран увек укљичен

Прикажи на закључаном екрану
Приказати на закључаном екрану

ПРИЈАВИ ПРОБЛЕМ
ПРИЈАВИТЕ ПРОБЛЕМ

Пријави се
Пријавите се

Где јести
Храна

За кампер возила
Камп

Сними путању
Снимање трасе

Додај место на OpenStreetMap
Додајte место на OpenStreetMap

Креирај нову листу
Креирање нове листе

Погледај на мапи
Погледајте на мапи

НАСТАВИ СНИМАЊЕ
НАСТАВИТИ СНИМАЊЕ

ЗАУСТАВИ БЕЗ СНИМАЊА
ЗАУСТАВИТИ БЕЗ СНИМАЊА

САЧУВАЈ
САЧУВАТИ

Заустави снимање путање
Зауставити снимање трасе

Путања је празна, нема шта да се сачува
Нема ништа за снимање

Рута од
Почетак

Рута до
Дестинација

Преузми мапе
Преузмите мапе

**Reply by dobridabar** [reply]
Hi,

Željko, some of your suggestions make sense to me, but I have doubts for the rest.

First, following are stilistic suggestions, and I think that the existing translations are closer to the terms that the original software uses (e.g. "рута" for navigation, "путања" for track recording), so I would prefer not to change them:

> Сними путању
> Снимање трасе
> 
> Заустави снимање путање
> Зауставити снимање трасе
> 
> Рута од
> Почетак
> 
> Рута до
> Дестинација
> 
> Где јести
> Храна
> 
> За кампер возила
> Камп
> 
> Путања је празна, нема шта да се сачува
> Нема ништа за снимање

Then, for the on/off switches I think it is the best if verb can be avoided becase it is not clear if that is a command to a program (which is typically expressed as informal imperative like "Омогући") or information from the program to the user (which is typically expressed in more formal manner, like formal imperative "Омогућите").

> Омогући снимање лог фајла
> Омогућите снимање лог фајла

How about "Снимање лог фајла"?

> Повећај слова на мапи
> Повећајте слова на мапи

How about "Увећана слова на мапи"?

> Упозорење на прекорачење брзине
> Упозорити код се прекорачи брзина

"Упозорење на прекорачење брзине" is already without a verb, so I think it is better.

> Сачувај мапе у
> Место за чување мапа

I agree.

> Држи укључен екран
> Екран увек укључен

I agree.

> Прикажи на закључаном екрану
> Приказати на закључаном екрану

How about "Приказ мапе на закључаном екрану"?


For command buttons and commands in general, I believe that informal imperative is a standard (e.g. "Креирај"), except when they lead to a longer process, and/or act as a hyperlinks.

Therefore, for the following four suggestions, which are commands from the user to the program, I think that first (existing) translation is better:

> Додај место на OpenStreetMap
> Додајte место на OpenStreetMap
>
> Креирај нову листу
> Креирање нове листе
> 
> НАСТАВИ СНИМАЊЕ
> НАСТАВИТИ СНИМАЊЕ
> 
> ЗАУСТАВИ БЕЗ СНИМАЊА
> ЗАУСТАВИТИ БЕЗ СНИМАЊА
> 
> САЧУВАЈ
> САЧУВАТИ
> 
> Преузми мапе
> Преузмите мапе

Then, for this one:

> Погледај на мапи
> Погледајте на мапи

How about "Прикажи на мапи", becase it is a command?


> ПРИЈАВИ ПРОБЛЕМ
> ПРИЈАВИТЕ ПРОБЛЕМ

I agree. "ПРИЈАВИТЕ ПРОБЛЕМ" is better, this is a longer process.


> Пријави се
> Пријавите се

I agree. "Пријавите се" is better, it is a title.

> Саобраћајне камере
> Прекршајне камере

I think first term is used more often. Google have 8 pages of results 
for the first term, 2 pages of results for the second.


Didn't create PR yet, waiting for your comments.

**Reply by ZeljkoBG** [reply]
I agree, do the best you can.

Term "Прекршајне камере" I came up with it in 2011. That's when I first installed cameras for all navigations. My base is used by everyone...
Saobraćajne kamere - is the wrong term. Because Traffic cameras are cameras that monitor traffic. Some are of normal shape and others are 360 ​​degrees. On them, the traffic police sees a crowd and sends a traffic officer to regulate the traffic.
Prekršajne kamere - is an accepted term because they correctly indicate the cameras that the navigation system needs to warn us about.

**Reply by dobridabar** [reply]
PR created: https://github.com/organicmaps/organicmaps/pull/10100

CC: @pastk

**Reply by ZeljkoBG** [reply]
@dobridabar
sr = Место за чувње мапа
A word mistake is missing - A - ...чувАње...

**Reply by dobridabar** [reply]
@ZeljkoBG, thanks. Fixed.

**Comment by zyphlar** [comment]
@dobridabar can you help with this? I've got a lot of other things going on right now

Jan 18, 2025 6:42:33 AM ZeljkoBG ***@***.***>:

**Reply by dobridabar** [reply]
Sure, I will create a PR later today.

**Comment by ZeljkoBG** [comment]
@dobridabar

Please replace the following text:
Атракција to Туристичке атракције or Знаменитости
Гориво to  Бензинске станице
Породичан одмор  to  Све за децу
Вода  to  Вода за пиће

**Reply by ZeljkoBG** [reply]
[zyphlar](https://github.com/zyphlar) create PR and correct this.

**Reply by dobridabar** [reply]
Sorry, I was busy for couple of days, will do this today or tomorrow.

**Reply by dobridabar** [reply]
"Water" is not same as "Drinking water", becase "Water" finds places with both drinking and non-drinking water.
Also "Gas" is not the same as "Gas station", becase "Gas" finds both gas stations and vending machines for fuel.

Will create a PR later today.

**Reply by zyphlar** [reply]
@ZeljkoBG continue working with dobridabar on this, I don't need to be involved. Thank you for your contributions!

**Reply by dobridabar** [reply]
PR: https://github.com/organicmaps/organicmaps/pull/10277

**Comment by ZeljkoBG** [comment]
I looked at the OSM tag for - Вода
Tag is [type.natural.water] - https://wiki.openstreetmap.org/wiki/Tag:natural%3Dwater (area) and of course it's not drinking water. Maybe it's clearer "Природна вода"

When I look at the OSM Tag on the Serbian maps, water sources are put under "natural.water" . I think the error for the source should be put on "drinking.water".
"Gas" - same

**Reply by dobridabar** [reply]
@ZeljkoBG, please reply ih the same thread, it is easier to follow.

My understanding is that you proposed to rename category search labels "Voda" (water) and "Gorivo" (gas) to "Voda za piće" (drinking water) and "Benzinska stanica" (gas station), respectively.

Regarding renaming "Voda" (water) to "Voda za piće" (drinking water), as I said it is not correct, because when you search category "Voda" you will get both drinking and non-drinking water. Test some region with more datiled maps then Serbia, like Germany, center of Berlin. You will clearly see that. And that has nothing to do with tag natural=water which is used to tag lakes, ponds, rivers, etc.

Regarding renaming "Gorivo" (gas) to "Benzinska stanica" (gas station), this is also incorrect, becase gas stations are places where you expect people, while searching for category "Gorivo" (gas) also returns self-service vending machines for fuel. See: https://wiki.openstreetmap.org/wiki/Tag:vending%3Dfuel

**Reply by ZeljkoBG** [reply]
It's all right.
I tried to explain to you"why I was wrong". But it doesn't matter.
